src/movies.js

```diff
@@ -1,15 +1,2102 @@
+    duration: "1h 24min",
+    genre: ["Animation", "Family", "Fantasy", "Musical", "Romance"],
+    rate: 8
+  },
+  {
+    title: "La battaglia di Algeri",
+    year: 1966,
+    director: "Gillo Pontecorvo",
+    duration: "2h 1min",
+    genre: ["Drama", "War"],
+    rate: 8.1
+  },
+  {
+    title: "Ah-ga-ssi",
+    year: 2016,
+    director: "Chan-wook Park",
+    duration: "2h 24min",
+    genre: ["Crime", "Drama", "Mystery", "Romance", "Thriller"],
+    rate: 8.1
+  },
+  {
+    title: "Pirates of the Caribbean: The Curse of the Black Pearl",
+    year: 2003,
+    director: "Gore Verbinski",
+    duration: "2h 23min",
+    genre: ["Action", "Adventure", "Fantasy"],
+    rate: 8
+  },
+  {
+    title: "PK",
+    year: 2014,
+    director: "Rajkumar Hirani",
+    duration: "2h 33min",
+    genre: ["Comedy", "Drama", "Fantasy", "Sci-Fi"],
+    rate: 8.2
+  },
+  {
+    title: "Dog Day Afternoon",
+    year: 1975,
+    director: "Sidney Lumet",
+    duration: "2h 5min",
+    genre: ["Biography", "Crime", "Drama", "Thriller"],
+    rate: 8
+  },
+  {
+    title: "Dead Poets Society",
+    year: 1989,
+    director: "Peter Weir",
+    duration: "2h 8min",
+    genre: ["Comedy", "Drama"],
+    rate: 8
+  }
+];
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies) {
+  const newArray = [...movies];
+  newArray.sort(function(objA, objB) {
+    if (objA.year < objB.year) {
+      return -1;
+    }
+    if (objA.year > objB.year) {
+      return 1;
+    }
+    if (objA.year === objB.year) {
+      if (objA.title < objB.title) {
+        return -1;
+      }
+      if (objA.title > objB.title) {
+        return 1;
+      }
+      if (objA.title === objB.title) {
+        return 0;
+      }
+      return 0;
+    }
+  });
+  return newArray;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movies) {
+  const newArray = movies.filter(function(genereObj, i, movies) {
+    if (genereObj.genre == "Drama") {
+      return true;
+    }
+  });
+  console.log("newArray", newArray);
+}
+howManyMovies(movies);
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(movies) {
+  movies.sort(function(objA, objB) {
+    if (objA.title < objB.title) {
+      return -1;
+    }
+    if (objA.title > objB.title) {
+      return 1;
+    }
+    if (objA.title === objB.title) {
+      return 0;
+    }
+  });
+
+  newList = movies.splice(0, 19);
+  console.log("newList :", newList);
+}
+orderAlphabetically(movies);
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies) {
+  const eAverage = movies.reduce(function(acc, averageObj, i, movies) {
+    let updatedMovies = acc + averageObj.rate;
+
+    return updatedMovies;
+  }, 0);
+
+  var num = eAverage / movies.length;
+  var n = num.toFixed(2);
```

Comment:
It is a good approach to use toFixed() but this method converts the number into a string. So in order to fix it you would have to converting it back into a number by wrapping everything into a parserFloat()